---
uint32 trajectory_size
